### PR TITLE
[ROK-685] List & ListItem Components

### PR DESCRIPTION
**Pull request by eynglv**
**Description:**
`List` and `ListItem` components that I'm using to update invoice settings UI. The list itself takes an optional `border` prop and required `children` props which are just `ListItems`. 

`ListItem` takes in `headerText`, `description`, `isHeader`, and `divider` props, all optional. You render primary actions by passing them in as `children`. A list item with no children is just a text row. 

**ScreenShots:**
Screen Shot 2022-06-02 at 11 13 06 AM
Screen Shot 2022-06-02 at 11 13 21 AM

### CODE REVIEW

File: src/components/List/List.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        listItem: {
+          [arg]: arg value,
+          ...
+        }
+      */
+    return val;
+  }, {});
+  return (
+    <Box width='500px'>
+      <List border='greyscale.500'>
```
Comment (smurmus) [inline src/components/List/List.stories.tsx:90]: do we want to be able to edit the border for the `List`? (will need to add another arg to the table in that case)

File: src/components/ListItem/ListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+export interface ListItemProps extends MuiListItemProps {
+  /**
+   * The main text of list item row
+   * @default undefined
+   * @optional
+   */
+  headerText?: string;
+  /**
+   * a sorta subheader or more text to supplement main header
+   * @optional i
```
Comment (smurmus) [inline src/components/ListItem/ListItem.tsx:19]: hmm I assume the "i" here is a typo? lol

File: src/components/ListItem/ListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  headerText,
+  description,
+  children,
+  divider = false,
+  isHeader,
+}: ListItemProps): JSX.Element => (
+  <Box>
+    <MuiListItem
+      alignItems='flex-start'
+      sx={{
```
Comment (smurmus) [inline src/components/ListItem/ListItem.tsx:46]: nit: single line

File: src/components/ListItem/ListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+const ListItem = ({
+  headerText,
+  description,
+  children,
+  divider = false,
+  isHeader,
+}: ListItemProps): JSX.Element => (
+  <Box>
+    <MuiListItem
+      alignItems='flex-start'
```
Comment (smurmus) [inline src/components/ListItem/ListItem.tsx:45]: nit: double quotes (hmm I should really get prettier set up here...)

File: src/components/ListItem/ListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+   */
+  isHeader?: boolean;
+}
+
+const ListItem = ({
+  headerText,
+  description,
+  children,
+  divider = false,
+  isHeader,
```
Comment (smurmus) [inline src/components/ListItem/ListItem.tsx:41]: since it's optional, do we need to do `isHeader = false`?

**Review by smurmus — APPROVED** [review]

File: components/index.d.ts [hunk 6]
```diff
@@ -11,6 +11,7 @@ import { TypographyProps as TypographyProps$1 } from '@mui/material/Typography';
 import { ImageListItemProps } from '@mui/material/ImageListItem';
 import { ImageListProps } from '@mui/material/ImageList';
 import { AccordionProps } from '@mui/material/Accordion';
+import { ListProps as ListProps$1, ListItemProps as ListItemProps$1 } from '@mui/material';
```
Comment (jlaurelli) [inline components/index.d.ts:14]: Well that's an interesting naming convention. I mean, MUI does do that underneath, but pretty funny to keep the convention on our side.

File: src/components/ListItem/ListItem.tsx (hunk 2 above)
Comment (eynglv) [reply src/components/ListItem/ListItem.tsx:19]: 😭

**Review by jlaurelli — APPROVED** [review]

File: src/components/ListItem/ListItem.tsx (hunk 4 above)
Comment (eynglv) [reply src/components/ListItem/ListItem.tsx:45]: ooh prettier will not let me do double quotes

File: src/components/List/List.stories.tsx (hunk 1 above)
Comment (smurmus) [reply src/components/List/List.stories.tsx:90]: as in, have a border toggle / input for color that we can pass into `list`, using similar logic to what you do for `list item` here (but with `category` of `list`)

File: src/components/ListItem/ListItem.tsx (hunk 4 above)
Comment (smurmus) [reply src/components/ListItem/ListItem.tsx:45]: oh sob. another thing for the backlog then....

**Comment by hearth-victoria** [comment]
@smurmus are you open to having a tag for this PR listed as "Hajimari"

**Pull request merged**